lib/src/main/java/com/thenewboston/api/bankapi/datasource/BankDataSource.kt

```diff
@@ -122,4 +123,22 @@ class BankDataSource @Inject constructor(private val networkClient: NetworkClien
             else -> Outcome.Success(blocks)
         }
     }
+
+    suspend fun fetchInvalidBlocks(): Outcome<InvalidBlockList> = makeApiCall(
+        call = { getInvalidBlocks() },
+        errorMessage = "Could not fetch list of invalid blocks"
+    )
+
+    private suspend fun getInvalidBlocks(): Outcome<InvalidBlockList> {
+        val invalidBlocks = networkClient.defaultClient
+            .get<InvalidBlockList>(BankAPIEndpoints.INVALID_BLOCKS_ENDPOINT)
+
+        return when {
+            invalidBlocks.results.isNullOrEmpty() -> Outcome.Error(
```

Comment:
For this use case, we can expect the invalid blocks list to be empty. A bank may not have any invalid blocks if they are just a recently new bank. 

So we can just check if results list is not null instead.

Reply:
I did want to write it at first as the endpoint never had invalid block list but then i changed my mind. Will fix it though, thanks!